appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProjectToolbar.java

```diff
@@ -36,9 +35,13 @@ public class ProjectToolbar extends Toolbar {
   private static final String WIDGET_NAME_SENDTONG = "Send to Gallery";
   private static final String WIDGET_NAME_LOGINTOGALLERY = "Login to Gallery";
 
-  private boolean isReadOnly;
+//  @UiField(provided=true)
+  public final boolean isReadOnly;
+//  @UiField(provided=true)
+  public final boolean galleryEnabled;
```

Comment:
It's not clear to me that these need to be public. It seems like they could be made private as they aren't referenced in the ui.xml file.

Reply:
> It's not clear to me that these need to be public. It seems like they could be made private as they aren't referenced in the ui.xml file.

It looks like with the commented code that that might been intended to be referenced by ui.xml. There are certainly more places where visibility of ui elements can be set using these variables passed to the UIBinder files. However, if it's working now, I'm inclined to leave it in this iteration and set the variables to private.